pkg/certs/k8s.go

```diff
@@ -409,41 +409,65 @@ func ensureMountedSecretsAreInSync(secret *v1.Secret, certDir string) error {
 func (pki PublicKeyInfrastructure) injectPublicKeyIntoMutatingWebhook(
 	ctx context.Context, client kubernetes.Interface, tlsSecret *v1.Secret,
 ) error {
-	config, err := client.AdmissionregistrationV1().MutatingWebhookConfigurations().Get(
-		ctx, pki.MutatingWebhookConfigurationName, metav1.GetOptions{})
-	if err != nil {
-		return err
-	}
+	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		config, err := client.AdmissionregistrationV1().MutatingWebhookConfigurations().Get(
+			ctx, pki.MutatingWebhookConfigurationName, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
 
-	for idx := range config.Webhooks {
-		config.Webhooks[idx].ClientConfig.CABundle = tlsSecret.Data["tls.crt"]
-	}
+		if len(config.Webhooks) == 0 {
+			return nil
+		}
+
+		oldConfig := config.DeepCopy()
 
-	_, err = client.AdmissionregistrationV1().
-		MutatingWebhookConfigurations().
-		Update(ctx, config, metav1.UpdateOptions{})
-	return err
+		for idx := range config.Webhooks {
+			config.Webhooks[idx].ClientConfig.CABundle = tlsSecret.Data["tls.crt"]
+		}
+
+		if reflect.DeepEqual(oldConfig.Webhooks, config.Webhooks) {
```

Comment:
I'm adding this check because we invoke this function during the scheduled maintenance, to avoid unnecessary updates